File: 필수문항/99_소수펠린드롬.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {

    public static boolean check(int num) {
        String word = Integer.toString(num);
        int lp = 0;
        int rp = word.length() - 1;
        while (lp <= rp) {
            if (word.charAt(lp) == word.charAt(rp)) {
                lp++;
                rp--;
            } else {
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        boolean[] sosu = new boolean[1003002];
        sosu[0] = true;
        sosu[1] = true;
        for (int i = 2; i <= Math.sqrt(1003002); i++) {
            int num = i * i;
            while (num < 1003002) {
                sosu[num] = true;
                num += i;
            }
        }
        for (int i = N; i < 1003002; i++) {
            if (!sosu[i] && check(i)) {
                System.out.println(i);
                break;
            }
        }

    }

}

File: 필수문항/99_소수펠린드롬.py
import sys
import math

input = sys.stdin.readline

N = int(input())

sosu = [True] * 1003002
sosu[0] = False
sosu[1] = False

for i in range(2, int(math.sqrt(1003002) + 1)):
    num = i**2
    while num < 1003002:
        sosu[num] = False
        num += i


def check(num):
    num = str(num)
    lp, rp = 0, len(num) - 1

    while True:
        if lp >= rp:
            return True

        if num[lp] == num[rp]:
            lp += 1
            rp -= 1
        else:
            return False


for i in range(N, 1003002):
    if sosu[i] and check(i):
        print(i)
        break
